evm/src/cpu/cpu_stark.rs

```diff
@@ -225,17 +225,17 @@ pub(crate) const MEM_CODE_CHANNEL_IDX: usize = 0;
 /// Index of the first general purpose memory channel.
 pub(crate) const MEM_GP_CHANNELS_IDX_START: usize = MEM_CODE_CHANNEL_IDX + 1;
 
-/// Recover the three components of an address, given a CPU row. The address
-/// is expected to be stored on the first memory channel, and the components
-/// are recovered as follows:
+/// Recover the three components of an address, given a CPU row and
+/// a provided memory channel index.
+/// The components are recovered as follows:
 ///
 /// - `context`, shifted by 2^64 (i.e. at index 2)
 /// - `segment`, shifted by 2^32 (i.e. at index 1)
 /// - `virtual`, not shifted (i.e. at index 0)
-pub(crate) const fn get_addr<T: Copy>(lv: &CpuColumnsView<T>) -> (T, T, T) {
-    let addr_context = lv.mem_channels[0].value[2];
-    let addr_segment = lv.mem_channels[0].value[1];
-    let addr_virtual = lv.mem_channels[0].value[0];
+pub(crate) const fn get_addr<T: Copy>(lv: &CpuColumnsView<T>, mem_channel: usize) -> (T, T, T) {
+    let addr_context = lv.mem_channels[mem_channel].value[2];
+    let addr_segment = lv.mem_channels[mem_channel].value[1];
+    let addr_virtual = lv.mem_channels[mem_channel].value[0];
```

Comment:
What I had in mind here was actually:

```rust
pub(crate) const fn get_addr<T: Copy, const IDX: usize>(lv: &CpuColumnsView<T>) -> (T, T, T) {
    let addr_context = lv.mem_channels[IDX].value[2];
    let addr_segment = lv.mem_channels[IDX].value[1];
    let addr_virtual = lv.mem_channels[IDX].value[0];
    (addr_context, addr_segment, addr_virtual)
}
```
…but I think what you did is more readable; `get_addr::<_, 0>(lv)` is just awkward to type.